fdbserver/worker.actor.cpp

```diff
@@ -745,17 +745,57 @@ ACTOR Future<Void> workerSnapCreate(WorkerSnapRequest snapReq, StringRef snapFol
 	return Void();
 }
 
-ACTOR Future<Void> monitorServerDBInfo( Reference<AsyncVar<Optional<ClusterControllerFullInterface>>> ccInterface, Reference<ClusterConnectionFile> connFile, LocalityData locality, Reference<AsyncVar<ServerDBInfo>> dbInfo ) {
+ACTOR Future<Void> monitorTraceLogIssues(Optional<Reference<AsyncVar<std::set<StringRef>>>> issues) {
+	state bool pingTimeout = false;
+	loop {
+		wait(delay(SERVER_KNOBS->TRACE_LOG_FLUSH_FAILURE_CHECK_INTERVAL_SECONDS));
+		Promise<Void> p;
+		pingTraceLogWriterThread(p);
```

Comment:
@ajbeamon Is this a safe usage of `Promise` across thread in general?

Reply:
Good question -- you'll need to use a ThreadSingleAssignmentVar/ThreadFuture, like is used in the Barrier action to be thread safe.

Reply:
Yeah I came up with a case where the future will never be trigger as a counter example to this usage.